File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        seqSum = nums[0];

        for i in range(1, len(nums)):
            if nums[i] == nums[i -1] +1:
                seqSum = seqSum + nums[i]
            else:
                break

        present = [False] * 1276

        for num in nums:
            present[num] = True

        while present[seqSum]:
            seqSum = seqSum + 1

        return seqSum

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.rs
impl Solution {
    pub fn missing_integer(nums: Vec<i32>) -> i32 {
        let mut seqSum:i32 = nums[0];

        for i in 1..nums.len() {
            if nums[i] == nums[i -1] +1 {
                seqSum += nums[i];
            }
            else {
                break;
            }
        }

        let mut present = vec![false; 1276];

        for &num in &nums {
            present[num as usize] = true;
        }

        while present[seqSum as usize] {
            seqSum += 1;
        }

        seqSum
    }
}